modules/react/text/lib/Text.tsx

```diff
@@ -1,76 +1,72 @@
 import * as React from 'react';
+import {base, system} from '@workday/canvas-tokens-web';
 import {createComponent} from '@workday/canvas-kit-react/common';
-// type changes no need
-import {type, CanvasTypeHierarchy, CanvasTypeVariants} from '@workday/canvas-kit-react/tokens';
-import {Box, BoxProps} from '@workday/canvas-kit-react/layout';
+import {createModifiers, createStyles, CSProps} from '@workday/canvas-kit-styling';
+import {mergeStyles, CommonStyleProps} from '@workday/canvas-kit-react/layout';
 
-export interface TextProps extends BoxProps {
+type TypeSize = 'large' | 'medium' | 'small';
+type TokenName = `${keyof typeof system.type}.${TypeSize}`;
+
+export interface TextProps extends CSProps, CommonStyleProps {
   /**
    * Type token as string with level and size separated by dot.
    * These values map to our [Canvas type levels](https://canvas.workday.com/tokens/type#type-styles).
    *
    * ```tsx
    * <Text typeLevel="body.small">Small body text</Text>
    * ```
-  */
-  typeLevel?: `${keyof CanvasTypeHierarchy}.${'large'|'medium'|'small'}`;
+   */
+  typeLevel?: TokenName;
+  children?: React.ReactNode;
   /**
    * Type variant token names: `error`, `hint` or `inverse`.
    *
    * ```tsx
    * <Text variant="error" typeLevel="subtext.large">Error text</Text>
    * ```
-  */
-  variant?: keyof CanvasTypeVariants;
+   */
+  variant?: 'error' | 'hint' | 'inverse';
 }
 
-/**
- * Function returns updated props by token and variant values
-*/
-const validateProps = ({typeLevel, variant, ...props}: TextProps) => {
-  let updatedProps: any = props;
-  const tokenPropNames = [
-    'color',
-    'fontFamily',
-    'fontSize',
-    'fontWeight',
-    'letterSpacing',
-    'lineHeight',
-  ];
+const createTypeStyles = (level: keyof typeof system.type, size: TypeSize, isHeading?: boolean) => {
+  return createStyles({
+    // @ts-ignore
+    ...system.type[level][size],
+    color: isHeading ? base.blackPepper400 : base.blackPepper300,
+  });
+};
```

Comment:
This code is not statically analyzable and will fail to build properly. `createStyles` cannot be inside a function call with variables. The build should throw an error in most cases, but at worst there is no error and some styles are simply skipped. Also the CSS Kit cannot be generated.

I'll have to work on an ESLint to catch these cases with explanations so we don't make this mistake.